lib/binary_to_decimal.rb

```diff
@@ -5,5 +5,25 @@
 # Calculate  and return the decimal value for this binary number using
 # the algorithm you devised in class.
 def binary_to_decimal(binary_array)
-  raise NotImplementedError
+  array = binary_array
+  power = 7
+  decimal_value = 0
+  array.each do |bit|
+    decimal_value += bit * (2 ** power)
+    power -= 1
+  end
+  return decimal_value
+  #raise NotImplementedError
+end
+
+def decimal_to_binary(decimal_number)
```

Comment:
This is actually reversing the binary digits, and your tests use numbers that read the same way backwards and forwards.